File: server-mh/src/Match/Handler/MatchPhases/StartPhase.ts
import { Phase } from "../Enums";
import { PhaseBase } from "./PhaseBase";
import { MatchContext } from "../Models/MatchContex";
import { START_DELAY_SECONDS, MATCH_TICK_RATE } from "../Consts";
import { Player } from "../Models/Player";

export class StartPhase extends PhaseBase {

    public override Start(context: MatchContext): void {
        context.state.tickCounter = START_DELAY_SECONDS * MATCH_TICK_RATE;
        context.broadcaster.LobbyStarted("Lobby Started");
    }
    public override Update(context: MatchContext): void {

        if (context.state.tickCounter <= 0) {

            context.logger.info(`Match started with ${context.state.players.length} players.players: ${context.state.players.map((p: Player) => p.userName).join(", ")}`);
            context.state.matchStarted = true;
            context.state.label.matchStarted = true;
            context.broadcaster.MatchStarted("Match Started");
            context.state.pendingPhase = Phase.Turn;
            return;
        }

        context.state.tickCounter--;


    }
}

File: Client-unity/Ludo/Assets/Scripts/OfflineMatchHandler/StartPhase.cs
using System.Collections.Generic;

namespace Ludo.Offline
{
    public class StartPhase : PhaseBase
    {
        public override void Start(MatchContext context)
        {
            context.State.TickCounter =
     MatchConstants.StartDelaySeconds *
     MatchConstants.MatchTickRate;
 
        }

        public override void Update(MatchContext context)
        {
            if (context.State.TickCounter <= 0)
            {
                context.State.Label.MatchStarted = true;
                context.CommandHandler.Enqueue(new MatchStartedCommand());
                var pieces = new List<PiecePositionDto>();

                foreach (var player in context.State.Players)
                {
                    foreach (var piece in player.Pieces)
                    {
                        pieces.Add(new PiecePositionDto
                        {
                            PlayerColor = player.Color,
                            PieceId = piece.Id,
                            CellIndex = piece.CurrentCell.Index
                        });
                    }
                }

                context.CommandHandler.Enqueue(
                    new PiecesPositionCommand(pieces)
                );
                context.State.PendingPhase = Phase.Turn;
                return;
            }

            context.State.TickCounter--;
        }
    }
}